driver/device.c

```diff
@@ -139,9 +139,19 @@ MagicConnectorIPV4NAT(_Inout_ WG_PEER *Peer, _Inout_ IPV4HDR *Hdr, _Inout_ UINT1
 
     *OldVal = (UINT16)(Hdr->Saddr & 0x0000FFFF);
 
-    KIRQL Irql = ExAcquireSpinLockShared(&Peer->EndpointLock);
-    UINT8 CurrentENatIndex = Peer->ENatIndex, CurrentONatIndex = Peer->ONatIndex;
-    ExReleaseSpinLockShared(&Peer->EndpointLock, Irql);
+    UINT8 CurrentPeerFirstOctet = __iso_volatile_load8(&Peer->PeerFirstOctet);
+    UINT8 CurrentENatIndex = __iso_volatile_load8(&Peer->ENatIndex);
+    UINT8 CurrentONatIndex = __iso_volatile_load8(&Peer->ONatIndex);
+
+    /*
+     * Capture the first outbound ping that happens from connector to access
+     * tier on start. This will help us figure out the first octet that we
+     * ought not NAT on the receive side.
+     */
+    if (CurrentPeerFirstOctet == 0 && Hdr->Protocol == IPPROTO_ICMP)
```

Comment:
should it check the ICMP type to make sure it's Echo Request (type 8)?

Reply:
Turns out this was true for the NAT case given some customer accounts; I went ahead and reordered things a bit to have the ping send right after we configure routes on connector. That takes advantage of the fact that the firewall rules are not present to allow inbound traffic so the connector initial ping seems to always make it through the interface first. I don't think we need this PR open anymore so I'll go ahead and merge.

I'll keep this idea noted for the future whenever we need another driver build. In addition to this ICMP type check, I think we should also just drop incoming traffic until `PeerFirstOctet` is set.